execution/leveldb.go

```diff
@@ -0,0 +1,125 @@
+package execution
+
+import (
+	"crypto/sha1"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/cnf/structhash"
+	"github.com/mesg-foundation/core/service"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+// LevelDB is a concrete implementation of the DB interface
+type LevelDB struct {
+	db *leveldb.DB
+}
+
+// New creates a new DB instance
+func New(path string) (DB, error) {
+	db, err := leveldb.OpenFile(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &LevelDB{db: db}, nil
+}
+
+// Create a record in the database to store this execution and returns the id
+// returns an error if any problem happen with the database
+// returns an error if inputs are invalid
+func (db *LevelDB) Create(service *service.Service, taskKey string, inputs map[string]interface{}, tags []string) (*Execution, error) {
+	task, err := service.GetTask(taskKey)
```

Comment:
GetTask + ReuireInputs logic dosen't belong to this package

Create should take Execution as parametr

So in that case it should only return erro

Reply:
In that case the `Complete` function will be the same and this package is only about the database and we need to create another package to store the logic of this execution package or we need to delegate this part to the api but this will make the api way more complicated.

The goal of this package is to be high level and manage this kind of verification.

We could create another package that only handles the database like that we can have it in the database folder but this package will be pretty much useless

Reply:
I was thinking about it and I think it shouldn't be there.

If you want to create Execution obj then do it with `execution.New`. Like every package this one should have execution.New (or NewFromService etc) and pass it to a database. 

Same for `complete` Exeuction object should have something like `func (e *Exection) Verify(outputKey, outputData) (error)` and this one should be called from `Complete`, but I still consider it (I mean verification) shouldn't be there at all.